File: nodejs/lib/SellerManager.js
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

/**
 * SellerManager - Manages seller data for marketplace
 *
 * Provides seller lookup and validation functionality using
 * mock seller data from JSON file.
 *
 * @class
 */
class SellerManager {
    static sellersData = null;

    /**
     * Loads seller data from JSON file
     *
     * @private
     * @static
     */
    static loadSellers() {
        if (this.sellersData) {
            return;
        }

        const jsonPath = path.join(__dirname, '..', 'data', 'mock-sellers.json');
        if (!fs.existsSync(jsonPath)) {
            throw new Error('Seller data file not found');
        }

        const json = fs.readFileSync(jsonPath, 'utf8');
        this.sellersData = JSON.parse(json);
    }

    /**
     * Gets all sellers
     *
     * @static
     * @returns {Array<Object>} Array of all seller objects
     */
    static getAllSellers() {
        this.loadSellers();
        return this.sellersData.sellers;
    }

    /**
     * Gets seller details by ID
     *
     * @static
     * @param {string} id - Seller identifier
     * @returns {Object|null} Seller object or null if not found
     */
    static getSellerById(id) {
        this.loadSellers();
        return this.sellersData.sellers.find(s => s.id === id) || null;
    }

    /**
     * Validates if a seller ID exists
     *
     * @static
     * @param {string} id - Seller identifier to validate
     * @returns {boolean} True if seller exists
     */
    static isValidSeller(id) {
        return this.getSellerById(id) !== null;
    }
}

export default SellerManager;

File: java/src/main/java/com/globalpayments/example/services/SellerManager.java
package com.globalpayments.example.services;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.globalpayments.example.models.Seller;
import com.globalpayments.example.models.SellersData;

import java.io.File;
import java.io.IOException;
import java.util.List;

/**
 * Manages seller data for marketplace operations.
 *
 * <p>Provides seller lookup and validation using mock data
 * loaded from JSON configuration file. Implements lazy loading
 * with thread-safe double-checked locking pattern.</p>
 *
 * @author Global Payments
 * @version 1.0
 */
public class SellerManager {
    private static SellersData sellersData;
    private static final Object lock = new Object();

    /**
     * Retrieves all registered sellers.
     *
     * @return List of all Seller objects
     */
    public static List<Seller> getAllSellers() {
        loadSellers();
        return sellersData.getSellers();
    }

    /**
     * Retrieves seller details by ID.
     *
     * @param id Seller identifier
     * @return Seller object with details, or null if not found
     */
    public static Seller getSellerById(String id) {
        loadSellers();
        return sellersData.getSellers().stream()
            .filter(s -> s.getId().equals(id))
            .findFirst()
            .orElse(null);
    }

    /**
     * Validates if a seller exists in the system.
     *
     * @param id Seller identifier to validate
     * @return true if seller exists, false otherwise
     */
    public static boolean isValidSeller(String id) {
        return getSellerById(id) != null;
    }

    /**
     * Loads seller data from JSON file using lazy initialization.
     * Thread-safe implementation using double-checked locking pattern.
     *
     * @throws RuntimeException if seller data file cannot be loaded
     */
    private static void loadSellers() {
        if (sellersData != null) {
            return;
        }

        synchronized (lock) {
            if (sellersData != null) {
                return;
            }

            try {
                ObjectMapper mapper = new ObjectMapper();
                File jsonFile = new File("data/mock-sellers.json");
                sellersData = mapper.readValue(jsonFile, SellersData.class);
            } catch (IOException e) {
                throw new RuntimeException("Failed to load seller data", e);
            }
        }
    }
}
